### PR TITLE
Terrain shader refactor 2

### PR DESCRIPTION
**Pull request by zeroyao**
### Purpose of this PR
- Rename *.ShaderVariables.hlsl -> *_Includes.hlsl
- TerrainLitGUI now collects custom shader properties to show their GUI under Terrain section. Made making custom terrain shaders less painful until we have the ShaderGraph for terrain shaders.
- Improved normal splat blending by blending the derivatives and convert to tangent space normal vector at the end.
- Added a normal pass in BaseMapGen shader for future VTC use.

---
### Release Notes
TBD.

---
### Testing status
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=terrain-shader-refactor&automation-tools_branch=master&unity_branch=trunk#

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [x] Tested UI multi-edition + Undo/Redo
- [x] C# and shader warnings (supress shader cache to see them)

**Automated Tests**: What did you setup?

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: Low

**Halo Effect**: None

---

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/Material/TerrainLit/TerrainLit.shader [hunk 1]
```diff
@@ -37,14 +37,14 @@ Shader "HDRP/TerrainLit"
...
 
         // HACK: GI Baking system relies on some properties existing in the shader ("_MainTex", "_Cutoff" and "_Color") for opacity handling, so we need to store our version of those parameters in the hard-coded name the GI baking system recognizes.
-        _MainTex("Albedo", 2D) = "white" {}
-        _Color("Color", Color) = (1,1,1,1)
+        [HideInInspector] _MainTex("Albedo", 2D) = "white" {}
+        [HideInInspector] _Color("Color", Color) = (1,1,1,1)
 
-        [ToggleUI] _SupportDecals("Support Decals", Float) = 1.0
-        [ToggleUI] _ReceivesSSR("Receives SSR", Float) = 1.0
+        [HideInInspector] [ToggleUI] _SupportDecals("Support Decals", Float) = 1.0
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Material/TerrainLit/TerrainLit.shader]: Is there a reasons to hide decals and receive SSR ? as currently they have an impact and work.

**Comment by sebastienlagarde** [comment]
Pleasethink to add link of katana in the PR

https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=terrain-shader-refactor&automation-tools_branch=master&unity_branch=trunk#

**Comment by sebastienlagarde** [comment]
I have merge master to have start from a green katana

File: com.unity.render-pipelines.high-definition/Runtime/Material/TerrainLit/TerrainLit.shader (hunk 1 above)
Comment (zeroyao) [reply com.unity.render-pipelines.high-definition/Runtime/Material/TerrainLit/TerrainLit.shader]: These UI are explicitly displayed by LitGUI (or maybe one of its base classes I don't remember for exact) so they are not affected by HideInInspector. However TerrainLit GUI now will iterate all the visible properties and display them, mostly for easier customization of TerrainLit-based shaders in user packages/projects.

**Pull request merged**